elements/linuxmodule/fromdevice.cc

```diff
@@ -389,14 +392,12 @@ FromDevice::got_skb(struct sk_buff *skb)
 {
     if (!_active)
 	return 0;		// 0 means not handled
-
-    Storage::index_type h = head(), t = tail(), nt = next_i(t);
-
-    if (nt != h) { /* ours */
-	skb = skb_share_check(skb, GFP_ATOMIC);
-	if (!skb)
-	    return 1;
-	assert(skb_shared(skb) == 0);
+    skb = skb_share_check(skb, GFP_ATOMIC);
+    if (!skb)
+        return 1;
+    assert(skb_shared(skb) == 0);
+    unsigned my_tail = take_tail();
```

Comment:
Also please don't say `my_tail`; `t` is just as good and requires fewer changes.